app/lib/models.dart

```diff
@@ -63,6 +66,23 @@ class Package extends db.ExpandoModel {
         .where((email) => email != lowerEmail)
         .toList();
   }
+
+  void updateVersion(db.Key newVersionKey) {
+    final Version newVersion = new Version.parse(newVersionKey.id);
```

Comment:
It is normally not encouraged to directly use `.id` because that is making assumptions. Instead if you passed in a `PackageVersion pv` object you could just use `pv.latestVersion` and `pv.latestSemanticVersion` (see the getters above). I would also add two new getters for the dev version.